com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Nodes/HDSampleBufferNode.cs

```diff
@@ -0,0 +1,207 @@
+                case BufferType.MotionVectors:
+                    AddSlot(new Vector2MaterialSlot(k_OutputSlotId, k_OutputSlotName, k_OutputSlotName, SlotType.Output, Vector2.zero, ShaderStageCapability.Fragment));
+                    channelCount = 2;
+                    break;
+                case BufferType.IsSky:
+                    AddSlot(new Vector1MaterialSlot(k_OutputSlotId, k_OutputSlotName, k_OutputSlotName, SlotType.Output, 0, ShaderStageCapability.Fragment));
+                    channelCount = 1;
+                    break;
+                case BufferType.PostProcessInput:
+                    AddSlot(new ColorRGBAMaterialSlot(k_OutputSlotId, k_OutputSlotName, k_OutputSlotName, SlotType.Output, Color.black, ShaderStageCapability.Fragment));
+                    channelCount = 4;
+                    break;
+            }
+
+            RemoveSlotsNameNotMatching(new[]
+            {
+                k_ScreenPositionSlotId,
+                k_SamplerInputSlotId,
+                k_OutputSlotId,
+            });
+        }
+
+        string GetFunctionName() => $"Unity_HDRP_SampleBuffer_{bufferType}_$precision";
+
+        public void GenerateNodeFunction(FunctionRegistry registry, GenerationMode generationMode)
+        {
+            // Preview SG doesn't have access to HDRP depth buffer
+            if (!generationMode.IsPreview())
+            {
+                registry.RequiresIncludePath("Packages/com.unity.render-pipelines.high-definition/Runtime/Material/NormalBuffer.hlsl");
+                registry.RequiresIncludePath("Packages/com.unity.render-pipelines.high-definition/Runtime/Material/Builtin/BuiltinData.hlsl");
+
+                registry.ProvideFunction(GetFunctionName(), s =>
+                {
+                    if (bufferType == BufferType.PostProcessInput)
+                    {
+                        // Declare post process input here because the property collector don't support TEXTURE_X type
+                        s.AppendLine($"TEXTURE2D_X({nameof(HDShaderIDs._CustomPostProcessInput)});");
+                        s.AppendLine($"SAMPLER(sampler{nameof(HDShaderIDs._CustomPostProcessInput)});");
+                    }
+
+                    s.AppendLine("$precision{1} {0}($precision2 uv, SamplerState samplerState)", GetFunctionName(), channelCount);
+                    using (s.BlockScope())
+                    {
+                        switch (bufferType)
+                        {
+                            case BufferType.NormalWorldSpace:
+                                s.AppendLine("uint2 pixelCoords = uint2(uv * _ScreenSize.xy);");
+                                s.AppendLine("NormalData normalData;");
+                                s.AppendLine("DecodeFromNormalBuffer(pixelCoords, normalData);");
+                                s.AppendLine("return normalData.normalWS;");
+                                break;
+                            case BufferType.Roughness:
```

Comment:
Rename BufferType.Roughness as BufferType.Smoothness and it must return the perceptual smoothness